src/librbd/io/Types.h

```diff
@@ -124,6 +124,27 @@ using striper::LightweightObjectExtents;
 typedef std::pair<uint64_t,uint64_t> Extent;
 typedef std::vector<Extent> Extents;
 
+struct ReadExtent {
+    const uint64_t offset;
+    const uint64_t length;
+    const LightweightBufferExtents buffer_extents;
+    ceph::bufferlist bl;
+    Extents extent_map;
+
```

Comment:
Nit: add default constructor (that initializes `offset` and `length`) and another that allows you to initialize the full object